include/circt/Dialect/FIRRTL/FIRRTLOps.h

```diff
@@ -24,27 +24,58 @@
 namespace circt {
 namespace firrtl {
 
+namespace direction {
+
+/// A type that indicates whether a port is an input or an output.
+enum Direction { Input, Output };
+
+/// The key in a module's attribute dictionary used to find the direction.
+const std::string attrKey = "portDirections";
```

Comment:
Please make this a `static const char * const attrKey = "portDirections";`

Yes, C++ is horrible.

Reply:
Fixed in rebase.